apis/python/src/tiledbsoma/_tdb_handles.py

```diff
@@ -368,7 +398,14 @@ def _opener(
         context: SOMATileDBContext,
         timestamp: int,
     ) -> clib.SOMAArray:
-        open_mode = clib.OpenMode.read if mode == "r" else clib.OpenMode.write
+        if mode == "r":
```

Comment:
nit: this is code is duplicated in SOMAGroupWrapper and SOMAArrayWrapper. Suggest pulling into a shared utility